pkg/kubelet/container/image_puller.go

```diff
@@ -99,7 +99,7 @@ func (puller *imagePuller) PullImage(pod *api.Pod, container *api.Container, pul
 		}
 	}
 
-	backOffKey := fmt.Sprintf("%s_%s", pod.Name, container.Image)
+	backOffKey := fmt.Sprintf("%s_%s_%s", pod.Name, pod.Namespace, container.Image)
```

Comment:
nit: use pod UID to avoid hitting the case of pod being recreated with the same name/namespace.